### PR TITLE
Added Guide to library

### PR DESCRIPTION
**Pull request by jasonzoladz**

### CODE REVIEW

**Comment by jasonzoladz** [comment]
Guide (sans types)

File: GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# A Guide to `purescript-routing`
+
+A couple notes upfront:
```
Comment (jdegoes) [inline GUIDE.md]: I think you need another newline here for this to render properly.

**Comment by jdegoes** [comment]
@jasonzoladz Looking good!

@cryogenian Please do the final review & merge when you're happy.

File: GUIDE.md (hunk 1 above)
Comment (garyb) [inline GUIDE.md]: Looks ok to me: https://github.com/jasonzoladz/purescript-routing/blob/master/GUIDE.md

Comment (jasonzoladz) [inline GUIDE.md]: Added newline and rebased.

**Comment by cryogenian** [comment]
Awesome! 
@jasonzoladz Thank you very much!

**Pull request merged**